### ISSUE TITLE
Problem with ttyrec handling

### ISSUE THREAD

**Issue by yobibyte** [facebookresearch/nle#114]
## 🐛 Bug

Creating two environments and resetting them in a row fails after 40e0cff0a84975e01200ccdf1afcde887520d1f3.

## To Reproduce

Add this to tests (in ```TestGymEnv```) and see FileNotFoundError at second reset:

```
    def test_double_reset(self, env_name, wizard):
        env = gym.make(env_name, wizard=wizard)
        _ = env.reset()
        env = gym.make(env_name, wizard=wizard)
        _ = env.reset()
```

## Expected behavior

Should not lead to FileNotFound.

## Environment

Collecting environment information...
NLE version: 0.5.2+46baabe
PyTorch version: 1.6.0
Is debug build: No
CUDA used to build PyTorch: None

OS: Mac OSX 10.15.7
GCC version: Could not collect
CMake version: version 3.18.4

Python version: 3.7
Is CUDA available: No
CUDA runtime version: No CUDA
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA

Versions of relevant libraries:
[pip3] libtorchbeast==0.0.14
[pip3] numpy==1.19.1
[pip3] torch==1.6.0
[pip3] torchvision==0.7.0
[conda] blas                      1.0                         mkl  
[conda] libtorchbeast             0.0.14                    dev_0    <develop>
[conda] mkl                       2019.4                      233  
[conda] mkl-service               2.3.0            py37hfbe908c_0  
[conda] mkl_fft                   1.1.0            py37hc64f4ea_0  
[conda] mkl_random                1.1.1            py37h959d312_0  
[conda] pytorch                   1.6.0                   py3.7_0    pytorch
[conda] torchvision               0.7.0                  py37_cpu    pytorch

**Comment by heiner** [comment]
Unclear how to fix this tbh: Each `gym.make` does a `chdir` and upon destruction it tries to `chdir` back to the previous directory. But with what you're doing here you destroy & create a new one in a single line.

We really should just have an issue that fixes the fact that libnethack even `chdir`s anywhere.

**Comment by yobibyte** [comment]
I'm not sure I understand. What is the intuition behind `nethack.close()` from the Nethack object? Never use this object again? If yes, shouldn't we create a new Nethack object? If not, should we have something like `open()` method which will recreate the vardir/ttyrec files? I guess `start_nle` is doing something like that?

**Comment by heiner** [comment]
`nethack.close()` does mean this object will never run again and we do create a new `nethack` object here. That's what happens in your second `env = gym.make(env_name, wizard=wizard)` call. It uses `weakref` magic to call `close()` on the `nethack` member of the old `env`, but only after the new env has been created. Before the new env gets created `cwd` is the vardir of the first env.

**Comment by condnsdmatters** [comment]
This has been fixed along with #132
